eos/utils/szego-polynomial.hh

```diff
@@ -15,90 +15,122 @@
+                    result[i] = std::sqrt(result[i]);
+                }
+
+                return result;
+            }
+
+        public:
+            SzegoPolynomial(const double & norm_measure, std::array<double, order_> && verblunsky_coefficients) :
+                _norm_measure(norm_measure),
+                _verblunsky_coefficients(verblunsky_coefficients),
+                _norms(_calculate_norms())
+            {
+            }
+
+            SzegoPolynomial(const SzegoPolynomial &) = default;
+            SzegoPolynomial(SzegoPolynomial &&) = default;
+            ~SzegoPolynomial() = default;
+            SzegoPolynomial & operator= (SzegoPolynomial &&) = default;
+            SzegoPolynomial & operator= (const SzegoPolynomial &) = default;
+
+            // Evaluate the normalized polynomials at on the real z axis, in the interval [-1, +1].
+            // Note that, contrary to the literature [S:2004B], we use an integral measure dmu, which
+            // yields \int dmu = `_norm_measure`, rather than the usual \int dmu = 1.
+            std::array<double, order_ + 1> operator() (const double & z) const
+            {
+                std::array<double, order_ + 1> phi;
+                std::array<double, order_ + 1> phi_star;
+
+                phi[0]      = 1.0;
+                phi_star[0] = 1.0;
+
+                // we use real-valued Verblunsky coefficients only.
+                for (unsigned n = 1 ; n <= order_ ; ++n)
+                {
+                    // cf. [S:2004B], eq. (1.4), p.2
+                    phi[n]      = z * phi[n - 1]  - _verblunsky_coefficients[n - 1] * phi_star[n - 1];
+
+                    // cf. [S:2004B], eqs. (1.4) and (1.5) in combination;
+                    phi_star[n] = phi_star[n - 1] - _verblunsky_coefficients[n - 1] * z * phi[n - 1];
+                }
+
+                std::array<double, order_ + 1> result;
+                for (unsigned n = 0 ; n <= order_ ; ++n)
+                {
+                    result[n] = phi[n] / _norms[n];
+                };
+
+                return result;
+            }
+
+            // Trivial generalization to real Verblunsky coefficients and complex z
+            std::array<complex<double>, order_ + 1> operator() (const complex<double> & z) const
+            {
+                std::array<complex<double>, order_ + 1> phi;
+                std::array<complex<double>, order_ + 1> phi_star;
+
+                phi[0]      = 1.0;
+                phi_star[0] = 1.0;
+
+                // we use real-valued Verblunsky coefficients only.
+                for (unsigned n = 1 ; n <= order_ ; ++n)
+                {
+                    // cf. [S:2004B], eq. (1.4), p.2
+                    phi[n]      = z * phi[n - 1]  - _verblunsky_coefficients[n - 1] * phi_star[n - 1];
+
+                    // cf. [S:2004B], eqs. (1.4) and (1.5) in combination;
+                    phi_star[n] = phi_star[n - 1] - _verblunsky_coefficients[n - 1] * z * phi[n - 1];
```

Comment:
Did you test this? I'm not so sure if this is correct. In particular, calculating `phi_star`.

Reply:
I did! I don't know why MacOs as trouble with that, it passes for the others